### PR TITLE
Create Icon component

### PR DESCRIPTION
**Pull request by dev387**
- Ref: [T76196](https://phabricator.nsoft.ba/T76196)

See also #73

### CODE REVIEW

File: src/components/Icon.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+.icon-wrapper {
+  position: relative;
+  font-size: 18px;
+}
+
+.label {
+  position: absolute;
+  left: 15px;
+  top: -5px;
+  font-size: 12px;
```
Comment (aldinezi) [inline src/components/Icon.vue]: please handle this with flex.

Comment (dev387) [reply src/components/Icon.vue]: What is wrong with this approach?

Comment (aldinezi) [reply src/components/Icon.vue]: This approach is OK if you can ensure that your component will always be used with these icon sizes. But as nsoft-components are intented to be versatile across products, many use cases should be considered. 
For example, imagine use case with larger icon size.

icons

Comment (dev387) [reply src/components/Icon.vue]: I can convert `px`s to `em`s and solve problem.

Comment (dev387) [reply src/components/Icon.vue]: If you are going to use something custom, you will probably made component yourself and  won't use this component

Comment (aldinezi) [reply src/components/Icon.vue]: It's not the point about using "something custom". As previously said, nsoft-components are intended to be versatile, if you could notice mixins like `typeable, shapeable..` etc. so each small component like button, checkbox can vary in size, shape, type, theme etc. 

What's the benefit of having a component like this, when I can always write a span and icon myself?

**Comment by aldinezi** [comment]
Also, I don't see how this component supports requirements listed in #73 .
`it should support variation attribute`, and support for `left and right for position inside button`.

File: src/components/Icon.vue (hunk 1 above)
Comment (dev387) [reply src/components/Icon.vue]: I don't see any benefits of this component anyway :disappointed:. You can always write something simple like this.

**Comment by stvedo** [comment]
> left and right for position inside button

I think that this should be handled in button component (or some other component that will be using Icon component)
I don't see how to align it to left or right since it doesn't have some wrapper.

**Comment by dev387** [comment]
It should be handled probably inside button component I guess?

**Comment by stvedo** [comment]
Variations should be supported.

File: src/components/Icon.vue (hunk 1 above)
Comment (dev387) [reply src/components/Icon.vue]: > I can convert `px`s to `em`s and solve problem.

Did you see this comment @aldinezi ?

**Comment by dev387** [comment]
I will add varioations and other mxins.

File: src/components/Icon.vue (hunk 1 above)
Comment (stvedo) [reply src/components/Icon.vue]: There is benefit that we don't have to create `NSoft font mixin` in all projects.
Just import component and display icon, and component will take care of fetching font, setting classes etc.

Comment (aldinezi) [reply src/components/Icon.vue]: I saw it. But I still think flex is a better idea. If you want to easily support on which side should label display, like top-left, top-right, bottom-right, bottom-left, then I would go with flex.

Comment (dev387) [reply src/components/Icon.vue]: Ok, I will convert to flex :heart:.

**Comment by aldinezi** [comment]
Please check the styling. When you have `variation-accent with-label` label is overflown, also icons in most cases are not horizontaly centered.

**Pull request merged**